x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -89,6 +102,9 @@
         dateTimeTestingFields.put(new Tuple<>("test_double", 1d), EsType.DOUBLE);
         dateTimeTestingFields.put(new Tuple<>("test_float", 1f), EsType.FLOAT);
         dateTimeTestingFields.put(new Tuple<>("test_keyword", "true"), EsType.KEYWORD);
+        if (isUnsignedLongSupported()) {
```

Comment:
Refactored.

[Offtopic remark]:
IMO this dedicated-methods refactoring enlarges the tests code base with little bottom line clarity. See `testGettingValidShortWithCasting` for instance -- anyone reading it would expect "short with casting" case to be covered there entirely -- that's what the method says. But then it's not: unsigned long is a special case, treated somewhere else (i.e. in the new `testGettingValidNumbersWithCastingFromUnsignedLong`). This might be clear "enough" now, but sometimes towards the end of 8.x dev cycle it'll likely no longer be the case.

And this could IMO be addressed by running both the driver and the QA tests of the same branch, rather than the mixed approach used currently.